test/state/precompiles_gmp.cpp

```diff
@@ -0,0 +1,30 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2025 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "precompiles_gmp.hpp"
+#include <gmp.h>
+#include <cassert>
+
+namespace evmone::state
+{
+void expmod_gmp(bytes_view base, bytes_view exp, bytes_view mod, uint8_t* output) noexcept
+{
+    mpz_t b, e, m, r;  // NOLINT(*-isolate-declaration)
+    mpz_inits(b, e, m, r, nullptr);
+    mpz_import(b, base.size(), 1, 1, 0, 0, base.data());
+    mpz_import(e, exp.size(), 1, 1, 0, 0, exp.data());
+    mpz_import(m, mod.size(), 1, 1, 0, 0, mod.data());
+    assert(mpz_sgn(m) != 0);
```

Comment:
I should document in the `expmod_xxx` interface that the `mod` is expected non-zero.